app/code/Magento/Catalog/Pricing/Price/MinimalTierPriceCalculator.php

```diff
@@ -58,6 +58,6 @@ public function getAmount(SaleableInterface $saleableItem)
 
         return $value === null
             ? null
-            : $this->calculator->getAmount($value, $saleableItem);
+            : $this->calculator->getAmount($value, $saleableItem, ['weee', 'weee_tax', 'tax']);
```

Comment:
The idea might be correct but the implementation is wrong since you add a knowledge about Weee module to Catalog. But Catalog knows nothing about Weee, it is Wee who depends to Catalog. So I think you may leave here mention about Tax but you need to extend this list from Weee module somehow.

Reply:
So as U mention I should remove 'weee' and 'weee_tax' from the array?

Reply:
Not so easy. As far as I understood `weee` and `weee_tax` are the parts of the fix. What I propose is to modify the solution the way your changes in `e/Magento/Catalog/Pricing/Price/MinimalTierPriceCalculator.php` get extensible and then leave there only mention about `tax` and extend it from module `Weee`. Does it make sense?

Reply:
Oh yes I understand it now. I'll make this change.

Reply:
I made this changing. Can you check it?

Reply:
@Hoszi Thank you fo the changes. It look better but it breaks [Magento Technical Gudelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html) (`2.6. Inheritance SHOULD NOT be used. Composition SHOULD be used for code reuse.`). Also you converted private variable to protected which also breaks Backward Compatibility (look [the guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)). I think in this case it would be better to add a new optional argument (e.g. `excludedAdjustments` array) to the class `app/code/Magento/Catalog/Pricing/Price/MinimalTierPriceCalculator.php` which will be configured in DI. In module `Catalog` you could add only one item `tax` and in the module `Weee` you could extend this list with `weee` and `weee_tax`. Doing so module `Catalog` doesn't need to know about `Weee`. Does it make sense?

Reply:
Yes. It make sanse. I'm on it. I will make this changes.